lib/internal/process/coverage.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const path = require('path');
+const { mkdirSync, writeFileSync } = require('fs');
+const inspector = require('inspector');
+const hasInspector = process.config.variables.v8_enable_inspector === 1 &&
+  require('internal/worker').isMainThread;
+
+let session;
+
+function writeCoverage() {
+  if (!hasInspector || !session) {
+    return;
+  }
+
+  const filename = `coverage-${process.pid}-${Date.now()}.json`;
```

Comment:
I think we need to account for the worker thread id here, and test interaction with it, as well…